app/main.py

```diff
@@ -1,3 +1,62 @@
+import functools
+from typing import Callable, Any
+
+
+def validate_operand(func: Callable) -> Callable:
+    @functools.wraps(func)
+    def wrapper(self: "Distance", other: ("Distance", int, float)) -> Any:
+        if isinstance(other, Distance):
+            return func(self, other.km)
+        elif isinstance(other, (int, float)):
+            return func(self, other)
+        else:
+            raise TypeError(f"Unsupported operand type for {func.__name__}")
+
+    return wrapper
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: (int, float)) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    @validate_operand
+    def __add__(self, other: ("Distance", int, float)) -> "Distance":
+        return Distance(self.km + other)
+
+    @validate_operand
+    def __iadd__(self, other: ("Distance", int, float)) -> "Distance":
+        self.km += other
+        return self
+
+    def __mul__(self, other: (int, float)) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        else:
+            raise TypeError("Unsupported operand type for *")
+
+    def __truediv__(self, other: int) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+        else:
+            raise TypeError("Unsupported operand type for /")
+
+    def __lt__(self, other: ("Distance", int, float)) -> bool:
+        return self.km < other
+
+    def __gt__(self, other: ("Distance", int, float)) -> bool:
+        return self.km > other
+
+    def __eq__(self, other: ("Distance", int, float)) -> bool:
+        return self.km == other
+
+    def __le__(self, other: ("Distance", int, float)) -> bool:
+        return not self > other
+
+    def __ge__(self, other: ("Distance", int, float)) -> bool:
+        return not self < other
```

Comment:
this implementation will require your decorator to be added to make this code work

Reply:
@Oleksl888 tests pass without the use of a decorator

Reply:
This ' not self < other' construct should have worked fine if we defined handling ints and floats in __lt__ , __gt__ , __eq__ methods but they are not. let me check and get back to you on that.

Reply:
I see - since you added your decorator on _add__ methods it extends it validation on comparison methods as well. My mistake